crates/lib/src/hex.rs

```diff
@@ -0,0 +1,130 @@
+use core::{fmt, ops, str};
+
+#[derive(Copy, Clone)]
+pub struct HexString<const N: usize> {
```

Comment:
I think this is useful enough to warrant a bit of commentary. Or maybe even a crate which can handle both fixed and dynamically sized data?

Also, can we not also use it for `Hash`?